packages/framework-provider-kubernetes-infrastructure/src/infrastructure/dapr-manager.ts

```diff
@@ -39,12 +47,27 @@ export class DaprManager {
   public async configureEventStore(): Promise<void> {
     const l = scopeLogger('configureEventStore', this.logger)
     l.debug('Starting to configure event store')
-    if (!fs.existsSync(this.daprComponentsPath)) {
+    const stateStoreFilePath = path.join(this.daprComponentsPath, this.stateStoreFileName)
+    if (fs.existsSync(stateStoreFilePath)) {
+      l.debug('statetore.yaml exists')
+      const stateStoreFileContent = fs.readFileSync(stateStoreFilePath).toString()
+      const yamlData = safeLoad(stateStoreFileContent) as stateStoreYaml
+      if (yamlData.metadata.annotations['booster/created']) {
+        l.debug('The state store is provisioned by Booster. Verifying that the state store is working on the cluster')
+        const templateValues: DaprTemplateValues = await this.ensureEventStoreIsReady()
+        await this.createDaprComponentFile(templateValues)
+        this.eventStorePassword = templateValues.eventStorePassword
+      } else {
+        l.debug('The state store is provisioned by the user. Getting statestore credentials')
+        //TODO: Get the credentials for the state store in K8s provider if the user provides us the statestore in Dapr. We need to get the DB_HOST, DB_USER and DB_PASS to pass it to the runtime
+        //Research how to get the eventStorePassword from the statestore file
+        //this.eventStorePassword has to be set
+      }
+    } else {
```

Comment:
I'd turn these `if`s around: Make them return early if the conditions are false. E.g.

```ts
if(!fs.existsSync(...)) {
  // debug and throw
}
// rest of the code